### DISCUSSION TITLE
Incidents get autodeleted?!

### DISCUSSION THREAD

**Discussion by FusselTV** [upptime/upptime#492] [Q&A] [answered]
I have successfully set up my status page but while testing I noticed, that the Incidents (GitHub Issues) get deleted after the site is back up. Is this normal behavior or have I missed something while configuring upptime?
https://github.com/FusselTV/status/issues/2 (the deleted issue, that was definitely not deleted by myself)
I would be very grateful about help!

**Comment by AnandChowdhary** [comment, marked as answer]
It often happens that Upptime might think a site is down when it's not actually down, so if it comes back up within a few minutes of being down, it's likely that it was not actually down in the first place, so we delete that issue automatically.

**Reply by AnandChowdhary** [reply]
The idea here was that otherwise you'll have a list of hundreds of phantom downtime issues :)

**Reply by FusselTV** [reply]
Would it be possible to add a config entry by default where you can find the value of how long a site needs to be down for a permanent entry? And a testing/debug mode that disables this autodeletion?

**Reply by vichaos** [reply]
> The idea here was that otherwise you'll have a list of hundreds of phantom downtime issues :)



> It often happens that Upptime might think a site is down when it's not actually down, so if it comes back up within a few minutes of being down, it's likely that it was not actually down in the first place, so we delete that issue automatically.

So how long does it needed to be down in order to have an incident report?

**Reply by AnandChowdhary** [reply]
This is implemented in https://github.com/upptime/uptime-monitor/blob/2c6717d37fe21fdd1d8731e3c52ceb5bf71bd084/src/summary.ts#L400. The current logic is to delete issues that were closed within 15 minutes of opening them (e.g., the ~very next run of Upptime marked it as up again, so it was probably never down in the first place) and it has only 1 comment (the default Upptime comment, meaning that no human commented to say that it's an actual issue).

You can add the configuration `skipDeleteIssues: false` in your Upptime configuration file to prevent this. This is implemented in https://github.com/upptime/uptime-monitor/blob/2c6717d37fe21fdd1d8731e3c52ceb5bf71bd084/src/summary.ts.
